addons/website_sale/models/product_misc.py

```diff
@@ -44,16 +44,19 @@ def clear_cache(self):
         website = self.env['website']
         website._get_pl_partner_order.clear_cache(website)
 
-    @api.model
-    def create(self, data):
-        if data.get('company_id') and not data.get('website_id'):
-            # l10n modules install will change the company currency, creating a
-            # pricelist for that currency. Do not use user's company in that
-            # case as module install are done with OdooBot (company 1)
-            self = self.with_context(default_company_id=data['company_id'])
-        res = super(ProductPricelist, self).create(data)
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('company_id') and not vals.get('website_id'):
+                # l10n modules install will change the company currency, creating a
+                # pricelist for that currency. Do not use user's company in that
+                # case as module install are done with OdooBot (company 1)
+                # YTI FIXME: The fix is not at the correct place
+                # It be set when we actually create the pricelist
+                self = self.with_context(default_company_id=vals['company_id'])
```

Comment:
I really don't understand that code.
What's the point of adding a default when you already have a value?

Reply:
It's all for the `_default_website` method above, to find the pricelist default website in the correct company.
But it could be done in a cleaner way IMHO (and won't work as expected on batch records creation if records are created in different companies)

Reply:
@rco-odoo This is a shitty fix that should be done way before ending into this create method, which is at the end of the call trace. 

Note that I didn't change anything here, I've just added a comment to ask for a cleanup. 

I would like to do it in a different task as I fear the side effects that it could bring.

Reply:
Looks like `_default_website` should be a compute that depends on field `company_id` :smiley: